dbms/src/Storages/Page/V3/BlobStore.cpp

```diff
@@ -307,7 +303,7 @@ void BlobStore::removePosFromStats(BlobFileId blob_id, BlobFileOffset offset, si
 {
     const auto & stat = blob_stats.blobIdToStat(blob_id);
     auto lock = stat->lock();
-    stat->removePosFromStat(offset, size);
+    stat->removePosFromStat(offset, size, lock);
```

Comment:
if the current thread already holds a `lock_guand` from L305, Does it still need to pass the lock into the sub-methods?

Reply:
how about changing `stat->lock()` back to hold the `lock` from thread. then we won't pass the lock_guard many functions...